gradle/publish.gradle

```diff
@@ -92,6 +92,16 @@ configure(javaProjects) { project ->
     }
 
     afterEvaluate {
+        // Common manifest attributes for all JAR files
```

Comment:
The default values for the manifest attributes for Java projects.

Reply:
As this is duplicated we can use the same trick as other shared configurations in this file and create a method that returns a closure that handles the configuration.  In the `ext` block above add the logic:

```
    configureJarManifestAttributes = { project ->
        { ->
            manifest.attributes['-exportcontents'] =  "*;-noimport:=true"
            manifest.attributes['Automatic-Module-Name'] =  project.group + '.' + project.archivesBaseName
            manifest.attributes['Build-Version'] =  project.gitVersion
            manifest.attributes['Bundle-Version'] =  project.version
            manifest.attributes['Bundle-Name'] =  project.archivesBaseName
            manifest.attributes['Bundle-SymbolicName'] =  project.group + '.' + project.archivesBaseName
        }
    }
}
```

Then you call it like so:

```
jar configureJarManifestAttributes(project)
```

The groovy syntax is weird but essentially we've created a method that takes a project returns a closure, which is then applied to the thing calling it `jar`.

Reply:
That works very well.